crates/rpcsrv/src/util.rs

```diff
@@ -1,7 +1,95 @@
+use std::{
+    borrow::Borrow,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+use dashmap::DashMap;
+
 /// Key used for the connections map.
 // TODO: Possibly per user connections?
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub(crate) struct ConnKey {
     pub(crate) ip: String,
     pub(crate) port: String,
 }
+
+pub struct ExpiringDashMap<K, V> {
```

Comment:
https://github.com/cobbinma/r-cache <- why not use something like this which has similar functionality and tests.

Reply:
This seemed trivial enough to just implement ourselves without either pulling in larger caching libraries like `moka`, or relying smaller unmaintained libraries like `r-cache`. 
I also wanted it to be self contained without having to spawn an external background job to clear the entries like is the case with r-cache.

Reply:
you do spawn the background job on line 48, so that seems roughly equivalent.

r-cache doesn't seem unmaintained: it's small and feature complete, there are no open issues, and they made changes in the last year. some things don't need much. 

r-cache also has tests which provide some extra confidence.

Reply:
if you feel strongly about using a library we can. This seemed trivial enough to not need one, so I didn't.